imap/message.c

```diff
@@ -908,7 +897,8 @@ static int message_parse_headers(struct msg *msg, struct body *body,
     }
 
     body->content_offset = msg->offset;
-    body->header_size = strlen(headers.s+1);
+    if (buf_len(&headers) >= body->boundary_size + 1)
+        body->header_size = buf_len(&headers) - body->boundary_size - 1;
 
```

Comment:
So, I assume body->header_size == 0 otherwise, and that is correct?

Also, unrelated to this PR, can we change the other instances of headers.s to buf_base(&headers)?

Reply:
yes, header_size is initialized to zero.

I had wanted to change the use of header.s to buf_base and I still can. But it will introduce more line noise, because we then should also switch to using a const char* instead of a `char*` that points into the buffer. I can do that, but it will change that code more and it's an egg dance already.

Reply:
Ok.  Keep it simple for now